cirq-core/cirq/contrib/acquaintance/executor.py

```diff
@@ -100,6 +106,47 @@ def optimization_at(
         )
 
 
+@cirq.transformer
+def strategy_executor(
+    circuit: 'cirq.Circuit',
+    *,
+    execution_strategy: ExecutionStrategy = None,
+    context: Optional['cirq.TransformerContext'] = None,
+) -> None:
+    """Executes an acquaintance strategy"""
+
+    if execution_strategy is None:
+        raise ArgumentError(execution_strategy, 'Execution strategy cannot be None')
+
+    mapping = execution_strategy.initial_mapping.copy()
```

Comment:
Do we need this copy?

Reply:
initial_mapping is just a LogicalMapping (a dictionary of integers). But it is mutated after every call to `op.gate.update_mapping(mapping, op.qubits)`. Since we're stateless I guess we can just pass around the same dictionary without creating copies like in StrategyExecutor.